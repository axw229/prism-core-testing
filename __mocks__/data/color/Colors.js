// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
