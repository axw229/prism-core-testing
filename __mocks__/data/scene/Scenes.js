// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import { type Scene, type Surface } from '../../../src/shared/types/Scene'

export const Scenes: Array<Scene> = [
  {
    "id": 1,
    "image": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_kitchen6_night?wid=1280}",
    "thumb": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_kitchen6_night?wid=300}",
    "width": 1200,
    "height": 725,
    "surfaces": [
      {
        "id": 1,
        "mask": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_kitchen6_night?wid=1280&req=object&opac=100&fmt=png-alpha&object=wall&color=FFFFFF}&fmt=png-alpha",
        "hitArea": "/prism/images/scenes/rooms/4/1.svg"
      },
      {
        "id": 2,
        "mask": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_kitchen6_night?wid=1280&req=object&opac=100&fmt=png-alpha&object=accent&color=FFFFFF}&fmt=png-alpha",
        "hitArea": "/prism/images/scenes/rooms/4/2.svg"
      },
      {
        "id": 3,
        "mask": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_kitchen6_night?wid=1280&req=object&opac=100&fmt=png-alpha&object=trim&color=FFFFFF}&fmt=png-alpha",
        "hitArea": "/prism/images/scenes/rooms/4/3.svg"
      }
    ]
  },
  {
    "id": 2,
    "image": "/prism/images/scenes/rooms/2/scene.jpg",
    "thumb": "/prism/images/scenes/rooms/2/scene.jpg",
    "width": 1200,
    "height": 725,
    "surfaces": [
      {
        "id": 1,
        "mask": "/prism/images/scenes/rooms/2/s1.png",
        "hitArea": "/prism/images/scenes/rooms/2/m1.svg"
      },
      {
        "id": 2,
        "mask": "/prism/images/scenes/rooms/2/s2.png",
        "hitArea": "/prism/images/scenes/rooms/2/m2.svg"
      },
      {
        "id": 3,
        "mask": "/prism/images/scenes/rooms/2/s3.png",
        "hitArea": "/prism/images/scenes/rooms/2/m3.svg"
      }
    ]
  },
  {
    "id": 3,
    "image": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom5?wid=1800}",
    "thumb": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom5?wid=300}",
    "width": 1200,
    "height": 725,
    "surfaces": [
      {
        "id": 1,
        "mask": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom5?wid=1200&req=object&opac=100&fmt=png-alpha&object=wall&color=FFFFFF}&fmt=png-alpha",
        "hitArea": "/prism/images/scenes/rooms/3/1.svg"
      },
      {
        "id": 2,
        "mask": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom5?wid=1200&req=object&opac=100&fmt=png-alpha&object=accent&color=FFFFFF}&fmt=png-alpha",
        "hitArea": "/prism/images/scenes/rooms/3/2.svg"
      },
      {
        "id": 3,
        "mask": "https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom5?wid=1200&req=object&opac=100&fmt=png-alpha&object=trim&color=FFFFFF}&fmt=png-alpha",
        "hitArea": "/prism/images/scenes/rooms/3/3.svg"
      }
    ]
  }
]

export function getScenes (): Array<Scene> {
  return cloneDeep(Scenes)
}

export function getScene (): Scene {
  return sample(getScenes())
}

export function getSurfaces (): Array<Surface> {
  return getScene().surfaces
}

export function getSurface (): Surface {
  return sample(getSurfaces())
}