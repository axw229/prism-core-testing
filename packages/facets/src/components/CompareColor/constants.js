export const baseClass = 'color__comparison'
export const containerClass = `${baseClass}__container`
export const containerHeaderClass = `${baseClass}__container__header`
export const containerHeaderButtonClass = `${baseClass}__container__header__button`
export const wrapperClass = `${baseClass}__wrapper`
export const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
export const buttonsClass = `${baseClass}__buttons`
export const buttonsVisibleClass = `${baseClass}__buttons--visible`
export const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
export const backgroundColorClass = `${baseClass}__background__color`
export const queueWrapperClass = `${baseClass}__queue__wrapper`
