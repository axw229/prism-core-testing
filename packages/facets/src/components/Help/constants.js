export const baseClass = `cvw-help`
export const wrapper = `${baseClass}__wrapper`
export const tabsContainer = `${baseClass}__tabs-container`
export const contentWrapper = `${baseClass}__content-wrapper`
export const helpContent = `${baseClass}__help-content`
export const helpContentHide = `${helpContent}--hide`
export const contentHeader = `${baseClass}__content-header`
export const contentDetails = `${baseClass}__content-details`
export const subContent = `${baseClass}__subcontent`
export const subContentHeader = `${baseClass}__subContent-header`
export const subContentDetails = `${baseClass}__subContent-details`
export const helpIcons = `${baseClass}__help-icons`
export const helpImages = `${baseClass}__help-images`
export const helpImagesMobile = `${baseClass}__help-images-mobile`
export const iconWrap = `${baseClass}__icon-wrap`
export const iconWrapUndoRedo = `${baseClass}__icon-wrap-undoredo`
export const iconInfo = `${baseClass}__icon-info`
export const secondIcon = `${baseClass}__second-icon`
export const SCROLL_SPEED = 500
