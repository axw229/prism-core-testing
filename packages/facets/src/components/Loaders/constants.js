// @flow
// the 5 colors used by loaders in their default state; taken from SW CVW circa 2019
export const DEFAULT_COLORS = [
  '#00b9e4',
  '#69be28',
  '#ffb612',
  '#e17000',
  '#b71234'
]
// how many colors there are (this is sort of magic, so don't touch it)
export const NUM_COLORS = DEFAULT_COLORS.length
// how many generated colors are allowed to be dark
export const NUM_DARKS = Math.floor(NUM_COLORS / 2)
// how many generated colors should be light
export const NUM_LIGHTS = NUM_COLORS - NUM_DARKS
// default brightness step between generated colors
export const GENERATED_COLOR_BRIGHTNESS_STEP = 10
// % of brightness spectrum edges considered to be too dark or too light
export const GENERATED_COLOR_EDGE_PCT = 0.1
