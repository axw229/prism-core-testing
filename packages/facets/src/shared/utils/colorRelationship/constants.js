export const FULL = 360
export const HALF = FULL / 2
export const QUARTER = HALF / 2
export const NUM_CLUSTERS = 12
export const CLUSTER_SIZE = FULL / NUM_CLUSTERS
