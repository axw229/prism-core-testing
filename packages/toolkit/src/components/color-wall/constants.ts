export const MAX_SWATCH_SIZE = 50
export const BASE_SWATCH_SIZE = 15
export const OUTER_SPACING = 20
export const MIN_SCROLLER_HEIGHT = 200
export const MAX_SCROLLER_HEIGHT = MIN_SCROLLER_HEIGHT * 3

export const MIN_BASE_SIZE = 14
export const MAX_BASE_SIZE = 45
export const ZOOMED_BASE_SIZE = 50
export const ACTIVE_SWATCH_SIZE = 150
const BLOOMED_SWATCH_MIN_SIZE = 70
export const BLOOMED_SWATCH_SIZES = new Array(4).fill(null).map((v, i, arr) => {
  return BLOOMED_SWATCH_MIN_SIZE + ((ACTIVE_SWATCH_SIZE - BLOOMED_SWATCH_MIN_SIZE) / arr.length) * (arr.length - 1 - i)
})

export const SWATCH_WIDTH_WRAP_THRESHOLD = 20 // will prefer wrapped view if unwrapped view target swatch size falls below threshold value

export const TITLE_SIZE_RATIOS = {
  1: 0.4, // l1 should be styled regular weight
  2: 0.4, // l2 should be styled bold weight
  3: 0.5 // l3 should be styled bold weight, and is larger than l1 and l2
}

export const TITLE_SIZE_MIN = 12
export const TITLE_SIZE_MAX = 16
