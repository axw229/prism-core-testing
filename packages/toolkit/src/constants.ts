export enum SCENE_TYPES {
  OBJECT = 'objects',
  ROOM = 'rooms',
  AUTOMOTIVE = 'automotive',
  FAST_MASK = 'fastmask'
}

export enum DRAG_TYPES {
  SWATCH = 'SWATCH'
}

export enum SCENE_VARIANTS {
  DAY = 'day',
  NIGHT = 'night',
  MAIN = 'main'
}

// ERRORS
export const SYSTEM_ERROR = 'SYSTEM_ERROR'

export enum BUTTON_POSITIONS {
  TOP = 'top',
  BOTTOM = 'bottom'
}

export const KEY_CODES = {
  KEY_CODE_ARROW_DOWN: 40,
  KEY_CODE_ARROW_LEFT: 37,
  KEY_CODE_ARROW_RIGHT: 39,
  KEY_CODE_ARROW_UP: 38,
  KEY_CODE_ENTER: 13,
  KEY_CODE_SPACE: 32,
  KEY_CODE_TAB: 9
}
