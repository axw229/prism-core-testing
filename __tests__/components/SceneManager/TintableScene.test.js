import React from 'react'
import renderer from 'react-test-renderer'

import TintableScene from 'src/components/SceneManager/TintableScene'
import TintableSceneOverlay from 'src/components/SceneManager/TintableSceneOverlay'
import TintableSceneHitArea from 'src/components/SceneManager/TintableSceneHitArea'
import TintableSceneSurface from 'src/components/SceneManager/TintableSceneSurface'
import { SCENE_TYPES } from 'constants/globals'
import { shallow } from 'enzyme'
import * as Colors from '__mocks__/data/color/Colors'
import { surfaces } from '__mocks__/data/scene/Scenes'

const getTintableScene = (props) => {
  let defaultProps = {
    width: 100,
    height: 100,
    sceneId: 1,
    type: SCENE_TYPES.ROOM,
    background: 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/TEST_SCENE?wid=1311}' // fully-qualified image so our snapshots don't change based on env vars
  }

  let newProps = Object.assign({}, defaultProps, props)
  return <TintableScene {...newProps} />
}

describe('<TintableScene />', () => {
  test('does not render without width and height', () => {
    expect(renderer.create(
      <TintableScene />
    ).toJSON()).toEqual(null)

    expect(renderer.create(
      <TintableScene width={100} />
    ).toJSON()).toEqual(null)

    expect(renderer.create(
      <TintableScene height={100} />
    ).toJSON()).toEqual(null)

    expect(renderer.create(
      getTintableScene({
        width: 0,
        height: 0 })
    ).toJSON()).toEqual(null)
  })

  test('does not render when render set to false', () => {
    expect(renderer.create(
      getTintableScene({ render: false })
    ).toJSON()).toEqual(null)
  })

  test('shows loading', () => {
    const scene = renderer.create(
      getTintableScene({
        loading: true
      })
    )

    // does it contain a TintableSceneOverlay and is it a LOADING type?
    expect(scene.root.findByType(TintableSceneOverlay).props.type).toEqual(TintableSceneOverlay.TYPES.LOADING)
  })

  test('shows error', () => {
    const scene = renderer.create(
      getTintableScene({
        error: true
      })
    )

    // does it contain a TintableSceneOverlay and is it an ERROR type?
    expect(scene.root.findByType(TintableSceneOverlay).props.type).toEqual(TintableSceneOverlay.TYPES.ERROR)
  })

  test('shows error overlay for hit area error', () => {
    const scene = shallow(
      getTintableScene({
        surfaces: surfaces,
        hitAreaError: false
      }))

    // no error at first
    expect(scene.find(TintableSceneOverlay).exists()).toBe(true)
    scene.instance().handleHitAreaLoadingError()
    // does it contain a TintableSceneOverlay and is it an ERROR type?
    expect(scene.instance().state.hitAreaError).toBe(true)
  })

  test('renders hit areas only for interactive scenes', () => {
    // interactive: true
    const scene = shallow(
      getTintableScene({
        surfaces: surfaces,
        interactive: true
      }))
    expect(scene.find(TintableSceneHitArea).exists()).toBe(true)

    // interactive: false
    const nonInteractiveScene = shallow(
      getTintableScene({
        surfaces: surfaces,
        interactive: false
      }))

    expect(nonInteractiveScene.find(TintableSceneHitArea).exists()).toBe(false)
  })

  test('calls parent prop to update surface colors when a surface is clicked and clickToPaint has a color', () => {
    const clickToPaintColor = Colors.getColor()
    const sceneId = 'testID'

    const scene = shallow(
      getTintableScene({
        surfaces: surfaces,
        clickToPaintColor: clickToPaintColor,
        sceneId: sceneId
      }))

    const instance = scene.instance()

    let spied = jest.spyOn(instance, 'updateSurfaceColor')

    // no TintableSceneSurfaces should render since none of the surface data contains color
    expect(scene.find(TintableSceneSurface).exists()).toBe(false)
    for (let i in surfaces) {
      instance.handleClickSurface(surfaces[i].id)
      expect(spied).toHaveBeenCalledWith(surfaces[i].id, clickToPaintColor)
    }
  })

  test('calls parent prop to update surface colors when a color is dropped on a surface', () => {
    const clickToPaintColor = jest.fn()
    const sceneId = 'testID'
    const scene = shallow(
      getTintableScene({
        surfaces: surfaces,
        clickToPaintColor: clickToPaintColor,
        sceneId: sceneId
      }))
    const instance = scene.instance()

    let spied = jest.spyOn(instance, 'updateSurfaceColor')

    // no TintableSceneSurfaces should render since none of the surface data contains color
    expect(scene.find(TintableSceneSurface).exists()).toBe(false)

    for (let i in surfaces) {
      const droppedColor = Colors.getColor()

      expect(instance.getTintColorBySurface(surfaces[i].id)).toBeUndefined()

      instance.handleColorDrop(surfaces[i].id, droppedColor)

      expect(spied).toHaveBeenCalledWith(surfaces[i].id, droppedColor)
    }
  })

  test('calls parent prop to update surface colors internal updateSurfaceColors is hit', () => {
    const onUpdateColor = jest.fn()
    const sceneId = 'testID'
    const scene = shallow(
      getTintableScene({
        surfaces: surfaces,
        onUpdateColor: onUpdateColor,
        sceneId: sceneId
      }))
    const instance = scene.instance()
    const color = Colors.getColor()

    instance.updateSurfaceColor(surfaces[0].id, color)

    expect(onUpdateColor).toHaveBeenCalledWith(sceneId, surfaces[0].id, color)
  })

  test('matches snapshot of scene with colored surfaces', () => {
    const filterSurfaces = surfaces.map(surface => {
      surface.color = Colors.getColor()
      return surface
    })

    const scene = shallow(
      getTintableScene({
        surfaces: filterSurfaces
      }))
    expect(scene).toMatchSnapshot()
  })

  test('renders for hovered surfaces when previewColor is active', () => {
    const scene = shallow(
      getTintableScene({
        surfaces: surfaces,
        previewColor: {
          hex: '#FFF'
        }
      }))

    // no TintableSceneSurfaces currently rendered
    expect(scene.find(TintableSceneSurface).exists()).toBe(false)

    // hover over first surface
    scene.instance().handleOver(surfaces[0].id)

    // one TintableSceneSurface should render
    expect(scene.find(TintableSceneSurface).exists()).toBe(true)

    // hover out first surface
    scene.instance().handleOut(surfaces[0].id)

    // zero TintableSceneSurfaces should render
    expect(scene.find(TintableSceneSurface).exists()).toBe(false)
  })
})
