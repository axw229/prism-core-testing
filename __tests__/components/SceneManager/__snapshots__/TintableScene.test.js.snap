// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TintableScene /> matches snapshot of scene with colored surfaces 1`] = `
<div
  className="prism-scene-manager__scene"
>
  <div
    className="prism-scene-manager__scene__inner"
    style={
      Object {
        "paddingTop": "100%",
      }
    }
  >
    <div
      className="prism-scene-manager__scene__svg-defs"
    >
      <TransitionGroup
        childFactory={[Function]}
        className="prism-scene-manager__transition__svg-defs"
        component="div"
      />
    </div>
    <div
      className="prism-scene-manager__scene__tint-wrapper"
    >
      <img
        className="prism-scene-manager__scene__natural"
        src="someImageURL"
      />
      <TransitionGroup
        childFactory={[Function]}
        className="prism-scene-manager__transition__colors"
        component="div"
      />
    </div>
    <LiveMessage
      aria-live="polite"
      message="undefined scene has been loaded"
    />
    <div
      className="prism-scene-manager__scene__hit-wrapper"
    >
      <DropTarget(TintableSceneHitArea)
        id={1}
        key="1"
        onClick={[Function]}
        onDrop={[Function]}
        onLoadingError={[Function]}
        onLoadingSuccess={[Function]}
        onOut={[Function]}
        onOver={[Function]}
      />
      <DropTarget(TintableSceneHitArea)
        id={2}
        key="2"
        onClick={[Function]}
        onDrop={[Function]}
        onLoadingError={[Function]}
        onLoadingSuccess={[Function]}
        onOut={[Function]}
        onOver={[Function]}
      />
      <DropTarget(TintableSceneHitArea)
        id={3}
        key="3"
        onClick={[Function]}
        onDrop={[Function]}
        onLoadingError={[Function]}
        onLoadingSuccess={[Function]}
        onOut={[Function]}
        onOver={[Function]}
      />
      <DropTarget(TintableSceneHitArea)
        id={4}
        key="4"
        onClick={[Function]}
        onDrop={[Function]}
        onLoadingError={[Function]}
        onLoadingSuccess={[Function]}
        onOut={[Function]}
        onOver={[Function]}
      />
      <DropTarget(TintableSceneHitArea)
        id={5}
        key="5"
        onClick={[Function]}
        onDrop={[Function]}
        onLoadingError={[Function]}
        onLoadingSuccess={[Function]}
        onOut={[Function]}
        onOver={[Function]}
      />
    </div>
    <TintableSceneOverlay
      type="LOADING"
    />
  </div>
</div>
`;
